### ISSUE TITLE
'"node"' is not recognized as an internal or external command, operable program or batch file.

### ISSUE THREAD

**Issue by oshihirii** [SublimeLinter/SublimeLinter#1835]
Sublime Text Version:  4121
Node:  v14.18.1
ESLint: v8.1.0

Originally posted here but asked to post elsewhere as it was not a Sublime Text issue:   

https://github.com/sublimehq/sublime_text/issues/5013#issuecomment-961857260

I was following directions on how to set up [SublimeLinter](https://packagecontrol.io/packages/SublimeLinter) from this [tutorial](https://nafeu.medium.com/automatic-code-linting-formatting-with-eslint-and-prettier-in-sublime-text-3-75c507cf9a0d).  

I have installed eslist (using `Git Bash` in `Windows Terminal Preview`):

    npm install -g eslint prettier
    which eslint prettier
    /c/Users/Me/AppData/Roaming/npm/eslint
    /c/Users/Me/AppData/Roaming/npm/prettier

Then did this:

```
Tools > Command Palette > Install Package > SublimeLinter  
Tools > Command Palette > Install Package > SublimeLinter-eslint  
Tools > Command Palette > Install Package > jsPrettier  
```

I have put user settings for SublimeLinter (`Preferences` > `Package Settings` > `Sublime Linter` > `Settings`) as:

```
// SublimeLinter Settings - User
{
  "linters": {
    "eslint": {
      "env": {
        "PATH": "/c/Users/Me/AppData/Roaming/npm/"
      },
      "args": ["--env=es6"]
    }
  }
}
```

And now just get this error:
```
SublimeLinter: #1 eslint app_es6.js ERROR:
==========================================
'"node"' is not recognized as an internal or external command,
operable program or batch file.
```
I've restarted Sublime Text several times (and the computer) and tried variations of that `PATH` value, but no luck.  

I use node all the time and can run:
```
$ node -v
v14.18.1
```

So `node` seems to be on my `PATH`, I double checked to make sure:

_System Variables_  
![system_variables](https://user-images.githubusercontent.com/5090307/140516282-6894d2fc-8287-4b8e-a9b6-5b0c1c48f631.png)

_I also tried adding the `nodejs` value to the `User Variables` section (but have removed that now)_    
![user_variables](https://user-images.githubusercontent.com/5090307/140516214-a2e507cf-c4b1-4a8d-9f58-6b349ce15d5e.png)

Looks like there was a similar issue posted back in 2017, but no solution there:

https://github.com/SublimeLinter/SublimeLinter/issues/529  

Reminds me of this:

https://xkcd.com/979/

**Comment by rchl** [comment]
Don't override thee env `PATH`.

**Comment by oshihirii** [comment]
Ah ok, have added related edit to post.  Will remove the entry in the user variable section when I am back on PC.  The issue was occurring before I added the value to the user variable section though.

**Comment by kaste** [comment]
I'm confused as to where the forward slashes come from.  Sure it looks like you're on Windows but `/c/Users/Me/AppData/Roaming/npm/eslint` is not typical.  

Doing 

```
      "env": {
        "PATH": "/c/Users/Me/AppData/Roaming/npm/"
      },
```

*should* be wrong as your replacing `PATH` with just that one path. Typically you want to append or prepend something to the `PATH`.  (Although I don't know why you need to set something manually here.)

For example, iirc

```
      "env": {
        "PATH": "/c/Users/Me/AppData/Roaming/npm/;$PATH"
      },
```

(Honestly I don't remember if `;` is the correct separator here.)

**Comment by oshihirii** [comment]
I have updated original post to provide more information.

I have removed what I added to SublimeLinter settings via:

`Preferences` > `Package Settings` > `Sublime Linter` > `Settings`

So it is now:
```js
// SublimeLinter Settings - User
{

}
```

I am no longer getting an error, but I am not sure if it is "working".  

When I have a JS file open, I am getting different colored bars (green, yellow), arrows (red) and a single dot (red) in the "gutter", but I only see any information when I hover over the red dot.  

Questions:  

- If this is "working" as it should, how can I see information about what the other errors are (green and yellow lines, red arrows etc)?    

- There doesn't seem to be any tool to "run eslint" on the file, it just seems to be happening - is that the default behaviour?  

![eslint_gutter_colors_no_messages](https://user-images.githubusercontent.com/5090307/140594516-6b84e721-3080-415f-80dd-693d26fa4cee.png)
